src/Helper/Uri.php

```diff
@@ -310,15 +310,40 @@ public function getPass(mixed $default = null): ?StringHelper
 
     /**
      * Returns TRUE if the host is valid.
-     * todo: Implement check for host URI segment
      *
      * @param string|null $host
      * @return boolean
      */
     public function checkHost(string $host = null): bool
     {
         if ($host === null) {
-            $host = $this->host;
+            $host = $this->getHost();
```

Comment:
Mhmm, I'm kinda unsure about it. I thought, this would be THE solution. 😅

During my tests for the port tests I had the issue, that with `$port = $this->port;` I got the error `HelperString: Can not convert to int` (or similar). That's the reason, why I've used the respective function and also did the same then for the host stuff.

So the same logic will fail with using `$port = $this->port;`. Not sure, we will have the same issue for `$host` - I haven't tested it there since it already failed for `$port`. 🙈

Reply:
Hmm, ok I'll think on this further for a bit. I'm not 100% positive, but I think port should always be a string, yes? If you want to check number-logic, it can always be type casted, e.g. `if ((int) $this->port <= 0) { ... }`.

Reply:
Since ports are numbers of a specific range (1-65535), I would expect that these are numbers (integers) and not strings. 🤔 

Yeah, I've tried it with type casting, but this also has thrown the the mentioned convert error for my PHPUnit tests (at least for the invalid ports).

Reply:
Yeah, type of integer makes sense to me. However, I believe the reasoning for using a string for port is that it's only ever used in a string context (e.g. in URI, printing, sending requests, etc all use string construct) , which also makes sense.

I've found usages of `port` as both `(int)` and `(string)`, so I think the solution here is to acknowledge that the framework can/does use both types for port, so we should intentionally be ignorant and just check that we can always get a valid `(int)` value out, even when stored as a string, but not require the internal type to be `(int)`, etc, just use `intval($this-port)` instead.

Usage references of `port`:
https://github.com/planetteamspeak/ts3phpframework/blob/5759ab6f4256382e83dfdb1c3424c9261d82420b/src/Transport/TCP.php#L54
https://github.com/planetteamspeak/ts3phpframework/blob/5759ab6f4256382e83dfdb1c3424c9261d82420b/src/Adapter/Adapter.php#L155
https://github.com/planetteamspeak/ts3phpframework/blob/5759ab6f4256382e83dfdb1c3424c9261d82420b/src/Helper/Uri.php#L68
https://github.com/planetteamspeak/ts3phpframework/blob/5759ab6f4256382e83dfdb1c3424c9261d82420b/src/Helper/Uri.php#L649

Reply:
Changed.